src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2546,18 +2677,9 @@ private bool ShouldInferDeclarationExpressionValEscape(BoundExpression argument,
             uint escapeTo = scopeOfTheContainingExpression;
 
             // collect all writeable ref-like arguments, including receiver
-            var receiverType = receiverOpt?.Type;
-            if (receiverType?.IsRefLikeOrAllowsRefLikeType() == true && !IsReceiverRefReadOnly(symbol))
-            {
-                // https://github.com/dotnet/roslyn/issues/73550:
-                // We do not have a test that demonstrates that the statement below makes a difference.
-                // If it is commented out, not a single test fails 
-                escapeTo = GetValEscape(receiverOpt, scopeOfTheContainingExpression);
```

Comment:
This code is indeed unnecessary. Thinking through the check for `IsReceiverRefReadOnly` is what caused me to dig into the `get / set` issues with indexers though.